### PR TITLE
Fixes #1285

### LINKED ISSUE #1285: Prison Hangar Doors Bugged
## Reproduction:

Basically, the 3 tiles to the right of the main blast door tile are able to be walked through. They don't block movement. Which means people or xenos can fling themselves off into space. Now, this is the RESEARCH hangar ones, not the ones at LZ1, those are working properly to block movement. 
Blast Door

### PR DESCRIPTION
**Pull request by Zostroll**
## About The Pull Request
Fixes #1285

## Why It's Good For The Game
Fixes a bug

## Changelog
:cl:
fix: Stops people from walking through shutters at research to walk on water.
/:cl:

### CODE REVIEW

File: code/game/machinery/doors/poddoor.dm [hunk 1]
```diff
@@ -282,7 +282,7 @@
 	var/obj/machinery/door/poddoor/filler_object/f2
 	icon = 'icons/obj/doors/1x2blast_hor.dmi'
 
-	New()
+	Initialize()
```
Comment (DominikPanic) [inline code/game/machinery/doors/poddoor.dm]: Convert these into absolute typepaths please (poke me on the Discord if you need help ^^)

File: code/game/machinery/doors/poddoor.dm [hunk 2]
```diff
@@ -282,7 +282,7 @@
 	var/obj/machinery/door/poddoor/filler_object/f2
 	icon = 'icons/obj/doors/1x2blast_hor.dmi'
 
-	New()
+	/obj/machinery/door/poddoor/two_tile_hor/Initialize()
```
Comment (spookydonut) [inline code/game/machinery/doors/poddoor.dm]:
```suggestion
/obj/machinery/door/poddoor/two_tile_hor/Initialize()
```

File: code/game/machinery/doors/poddoor.dm [hunk 3]
```diff
@@ -282,7 +282,7 @@
...
 		..()
```
Comment (spookydonut) [inline code/game/machinery/doors/poddoor.dm]:
```suggestion
		. = ..()
```

File: code/game/machinery/doors/poddoor.dm [hunk 4]
```diff
@@ -303,7 +304,7 @@
 	var/obj/machinery/door/poddoor/filler_object/f2
 	icon = 'icons/obj/doors/1x2blast_vert.dmi'
 
-	New()
+	/obj/machinery/door/poddoor/two_tile_ver/Initialize()
```
Comment (spookydonut) [inline code/game/machinery/doors/poddoor.dm]:
```suggestion
/obj/machinery/door/poddoor/two_tile_ver/Initialize()
```

File: code/game/machinery/doors/poddoor.dm [hunk 5]
```diff
@@ -303,7 +304,7 @@
...
 		..()
```
Comment (spookydonut) [inline code/game/machinery/doors/poddoor.dm]:
```suggestion
	. = ..()
```

**Comment by Zostroll** [comment]
Yeah at @spookydonut  I made some changes on my phone and I can't fix indentation on my phone so I was gonna fix it now..

**Comment by Zostroll** [comment]
and now I am failing the checks...

File: code/game/machinery/doors/poddoor.dm [hunk 6]
```diff
@@ -282,16 +282,17 @@
 	var/obj/machinery/door/poddoor/filler_object/f2
 	icon = 'icons/obj/doors/1x2blast_hor.dmi'
 
-	New()
-		..()
+/obj/machinery/door/poddoor/two_tile_hor/Initialize()
+		. = ..()
```
Comment (spookydonut) [inline code/game/machinery/doors/poddoor.dm]:
```suggestion
	. = ..()
```

File: code/game/machinery/doors/poddoor.dm [hunk 7]
```diff
@@ -282,16 +282,17 @@
...
 		f1 = new/obj/machinery/door/poddoor/filler_object (loc)
```
Comment (spookydonut) [inline code/game/machinery/doors/poddoor.dm]:
```suggestion
	f1 = new/obj/machinery/door/poddoor/filler_object (loc)
```

File: code/game/machinery/doors/poddoor.dm [hunk 8]
```diff
@@ -282,16 +282,17 @@
...
 		f2 = new/obj/machinery/door/poddoor/filler_object (get_step(src,EAST))
```
Comment (spookydonut) [inline code/game/machinery/doors/poddoor.dm]:
```suggestion
	f2 = new/obj/machinery/door/poddoor/filler_object (get_step(src,EAST))
```

File: code/game/machinery/doors/poddoor.dm [hunk 9]
```diff
@@ -282,16 +282,17 @@
...
 		f1.density = density
```
Comment (spookydonut) [inline code/game/machinery/doors/poddoor.dm]:
```suggestion
	f1.density = density
```

File: code/game/machinery/doors/poddoor.dm [hunk 10]
```diff
@@ -282,16 +282,17 @@
...
 		f2.density = density
```
Comment (spookydonut) [inline code/game/machinery/doors/poddoor.dm]:
```suggestion
	f2.density = density
```

File: code/game/machinery/doors/poddoor.dm [hunk 11]
```diff
@@ -282,16 +282,17 @@
...
 		f1.SetOpacity(opacity)
```
Comment (spookydonut) [inline code/game/machinery/doors/poddoor.dm]:
```suggestion
	f1.SetOpacity(opacity)
```

File: code/game/machinery/doors/poddoor.dm [hunk 12]
```diff
@@ -282,16 +282,17 @@
...
 		f2.SetOpacity(opacity)
```
Comment (spookydonut) [inline code/game/machinery/doors/poddoor.dm]:
```suggestion
	f2.SetOpacity(opacity)
```

File: code/game/machinery/doors/poddoor.dm [hunk 13]
```diff
@@ -282,16 +282,17 @@
...
 
-	Destroy()
+	
+/obj/machinery/door/poddoor/two_tile_hor/Destroy()
 		qdel(f1)
```
Comment (spookydonut) [inline code/game/machinery/doors/poddoor.dm]:
```suggestion
	qdel(f1)
```

File: code/game/machinery/doors/poddoor.dm [hunk 14]
```diff
@@ -282,16 +282,17 @@
...
 		f1 = null
```
Comment (spookydonut) [inline code/game/machinery/doors/poddoor.dm]:
```suggestion
	f1 = null
```

File: code/game/machinery/doors/poddoor.dm [hunk 15]
```diff
@@ -282,16 +282,17 @@
...
 		qdel(f2)
```
Comment (spookydonut) [inline code/game/machinery/doors/poddoor.dm]:
```suggestion
	qdel(f2)
```

File: code/game/machinery/doors/poddoor.dm [hunk 16]
```diff
@@ -303,16 +304,17 @@
 	var/obj/machinery/door/poddoor/filler_object/f2
 	icon = 'icons/obj/doors/1x2blast_vert.dmi'
 
-	New()
-		..()
+/obj/machinery/door/poddoor/two_tile_ver/Initialize()
+	. = ..()
 		f1 = new/obj/machinery/door/poddoor/filler_object (loc)
```
Comment (spookydonut) [inline code/game/machinery/doors/poddoor.dm]:
```suggestion
	f1 = new/obj/machinery/door/poddoor/filler_object (loc)
```

File: code/game/machinery/doors/poddoor.dm [hunk 17]
```diff
@@ -303,16 +304,17 @@
...
 		f2 = new/obj/machinery/door/poddoor/filler_object (get_step(src,NORTH))
```
Comment (spookydonut) [inline code/game/machinery/doors/poddoor.dm]:
```suggestion
	f2 = new/obj/machinery/door/poddoor/filler_object (get_step(src,NORTH))
```

File: code/game/machinery/doors/poddoor.dm [hunk 18]
```diff
@@ -303,16 +304,17 @@
...
 		f1.density = density
```
Comment (spookydonut) [inline code/game/machinery/doors/poddoor.dm]:
```suggestion
	f1.density = density
```

File: code/game/machinery/doors/poddoor.dm [hunk 19]
```diff
@@ -303,16 +304,17 @@
...
 		f2.density = density
```
Comment (spookydonut) [inline code/game/machinery/doors/poddoor.dm]:
```suggestion
	f2.density = density
```

File: code/game/machinery/doors/poddoor.dm [hunk 20]
```diff
@@ -303,16 +304,17 @@
...
 		f1.SetOpacity(opacity)
```
Comment (spookydonut) [inline code/game/machinery/doors/poddoor.dm]:
```suggestion
	f1.SetOpacity(opacity)
```

File: code/game/machinery/doors/poddoor.dm [hunk 21]
```diff
@@ -303,16 +304,17 @@
...
 		f2.SetOpacity(opacity)
```
Comment (spookydonut) [inline code/game/machinery/doors/poddoor.dm]:
```suggestion
	f2.SetOpacity(opacity)
```

File: code/game/machinery/doors/poddoor.dm [hunk 22]
```diff
@@ -303,16 +304,17 @@
...
 
-	Destroy()
+	
+/obj/machinery/door/poddoor/two_tile_ver/Destroy()
 		qdel(f1)
```
Comment (spookydonut) [inline code/game/machinery/doors/poddoor.dm]:
```suggestion
	qdel(f1)
```

File: code/game/machinery/doors/poddoor.dm [hunk 23]
```diff
@@ -303,16 +304,17 @@
...
 		f1 = null
```
Comment (spookydonut) [inline code/game/machinery/doors/poddoor.dm]:
```suggestion
	f1 = null
```

File: code/game/machinery/doors/poddoor.dm [hunk 24]
```diff
@@ -303,16 +304,17 @@
...
 		qdel(f2)
```
Comment (spookydonut) [inline code/game/machinery/doors/poddoor.dm]:
```suggestion
	qdel(f2)
```

File: code/game/machinery/doors/poddoor.dm [hunk 25]
```diff
@@ -325,8 +327,9 @@
 	var/obj/machinery/door/poddoor/filler_object/f3
 	var/obj/machinery/door/poddoor/filler_object/f4
 	icon = 'icons/obj/doors/1x4blast_hor.dmi'
+	resistance_flags = UNACIDABLE|INDESTRUCTIBLE
 
-	New()
+	/obj/machinery/door/poddoor/four_tile_hor/Initialize()
```
Comment (spookydonut) [inline code/game/machinery/doors/poddoor.dm]:
```suggestion
/obj/machinery/door/poddoor/four_tile_hor/Initialize()
```

File: code/game/machinery/doors/poddoor.dm [hunk 26]
```diff
@@ -325,8 +327,9 @@
...
 		..()
```
Comment (spookydonut) [inline code/game/machinery/doors/poddoor.dm]:
```suggestion
	. = ..()
```

File: code/game/machinery/doors/poddoor.dm [hunk 27]
```diff
@@ -325,8 +327,9 @@
...
 		f1 = new/obj/machinery/door/poddoor/filler_object (loc)
```
Comment (spookydonut) [inline code/game/machinery/doors/poddoor.dm]:
```suggestion
	f1 = new/obj/machinery/door/poddoor/filler_object (loc)
```

File: code/game/machinery/doors/poddoor.dm [hunk 28]
```diff
@@ -325,8 +327,9 @@
...
 		f2 = new/obj/machinery/door/poddoor/filler_object (get_step(f1,EAST))
```
Comment (spookydonut) [inline code/game/machinery/doors/poddoor.dm]:
```suggestion
	f2 = new/obj/machinery/door/poddoor/filler_object (get_step(f1,EAST))
```

File: code/game/machinery/doors/poddoor.dm [hunk 29]
```diff
@@ -341,7 +344,8 @@
 		f4.SetOpacity(opacity)
 		f3.SetOpacity(opacity)
 
-	Destroy()
+	
+/obj/machinery/door/poddoor/four_tile_hor/Destroy()
 		qdel(f1)
```
Comment (spookydonut) [inline code/game/machinery/doors/poddoor.dm]:
```suggestion
	qdel(f1)
```

File: code/game/machinery/doors/poddoor.dm [hunk 30]
```diff
@@ -341,7 +344,8 @@
...
 		f1 = null
```
Comment (spookydonut) [inline code/game/machinery/doors/poddoor.dm]:
```suggestion
	f1 = null
```

File: code/game/machinery/doors/poddoor.dm [hunk 31]
```diff
@@ -341,7 +344,8 @@
...
 		qdel(f2)
```
Comment (spookydonut) [inline code/game/machinery/doors/poddoor.dm]:
```suggestion
	qdel(f2)
```

File: code/game/machinery/doors/poddoor.dm [hunk 32]
```diff
@@ -360,7 +364,7 @@
 	icon = 'icons/obj/doors/1x4blast_vert.dmi'
 	resistance_flags = UNACIDABLE|INDESTRUCTIBLE
 
-	New()
+	/obj/machinery/door/poddoor/four_tile_ver/Initialize()
```
Comment (spookydonut) [inline code/game/machinery/doors/poddoor.dm]:
```suggestion
/obj/machinery/door/poddoor/four_tile_ver/Initialize()
```

File: code/game/machinery/doors/poddoor.dm [hunk 33]
```diff
@@ -360,7 +364,7 @@
...
 		..()
```
Comment (spookydonut) [inline code/game/machinery/doors/poddoor.dm]:
```suggestion
	. = ..()
```

File: code/game/machinery/doors/poddoor.dm [hunk 34]
```diff
@@ -360,7 +364,7 @@
...
 		f1 = new/obj/machinery/door/poddoor/filler_object (loc)
```
Comment (spookydonut) [inline code/game/machinery/doors/poddoor.dm]:
```suggestion
	f1 = new/obj/machinery/door/poddoor/filler_object (loc)
```

File: code/game/machinery/doors/poddoor.dm [hunk 35]
```diff
@@ -360,7 +364,7 @@
...
 		f2 = new/obj/machinery/door/poddoor/filler_object (get_step(f1,NORTH))
```
Comment (spookydonut) [inline code/game/machinery/doors/poddoor.dm]:
```suggestion
	f2 = new/obj/machinery/door/poddoor/filler_object (get_step(f1,NORTH))
```

File: code/game/machinery/doors/poddoor.dm [hunk 36]
```diff
@@ -375,7 +379,8 @@
 		f4.SetOpacity(opacity)
```
Comment (spookydonut) [inline code/game/machinery/doors/poddoor.dm]:
```suggestion
	f4.SetOpacity(opacity)
```

File: code/game/machinery/doors/poddoor.dm [hunk 37]
```diff
@@ -375,7 +379,8 @@
...
 		f3.SetOpacity(opacity)
```
Comment (spookydonut) [inline code/game/machinery/doors/poddoor.dm]:
```suggestion
	f3.SetOpacity(opacity)
```

File: code/game/machinery/doors/poddoor.dm [hunk 38]
```diff
@@ -375,7 +379,8 @@
...
 
-	Destroy()
+	
+/obj/machinery/door/poddoor/four_tile_ver/Destroy()
 		qdel(f1)
```
Comment (spookydonut) [inline code/game/machinery/doors/poddoor.dm]:
```suggestion
	qdel(f1)
```

File: code/game/machinery/doors/poddoor.dm [hunk 39]
```diff
@@ -375,7 +379,8 @@
...
 		f1 = null
```
Comment (spookydonut) [inline code/game/machinery/doors/poddoor.dm]:
```suggestion
	f1 = null
```

File: code/game/machinery/doors/poddoor.dm [hunk 40]
```diff
@@ -375,7 +379,8 @@
...
 		qdel(f2)
```
Comment (spookydonut) [inline code/game/machinery/doors/poddoor.dm]:
```suggestion
	qdel(f2)
```

File: code/game/machinery/doors/poddoor.dm [hunk 41]
```diff
@@ -282,42 +281,42 @@
...
+	f1.density = density
+	f2.density = density
+	f1.SetOpacity(opacity)
+	f2.SetOpacity(opacity)
+
+/obj/machinery/door/poddoor/two_tile_hor/Destroy()
+	qdel(f1)
+	f1 = null
+	qdel(f2)
+	f2 = null
```
Comment (DominikPanic) [inline code/game/machinery/doors/poddoor.dm]: QDEL_NULL(f2) etc would be better here.

Comment (Zostroll) [reply code/game/machinery/doors/poddoor.dm]: Is that what you meant?

**Pull request merged**